pull-kubernetes-unit flake TestEnsurePIPTagged

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

#### Which jobs are flaking:
https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-unit

#### Which test(s) are flaking:
      "k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure TestEnsurePIPTagged/ensurePIPTagged_should_ensure_the_pip_is_tagged_as_configured": 8,
      "k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure TestEnsurePIPTagged": 8,


#### Testgrid link:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/104652/pull-kubernetes-unit/1433338947647311872
failed exmaple

#### Reason for failure:
```
=== RUN   TestEnsurePIPTagged/ensurePIPTagged_should_ensure_the_pip_is_tagged_as_configured
W0902 08:30:50.869224   90701 azure_utils.go:135] parseTags: error when parsing key-value pair e-, would ignore this one
W0902 08:30:50.869276   90701 azure_utils.go:135] parseTags: error when parsing key-value pair -f, would ignore this one
W0902 08:30:50.869303   90701 azure_utils.go:130] parseTags: error when parsing key-value pair ghi, would ignore this one
    azure_loadbalancer_test.go:3842: 
        	Error Trace:	azure_loadbalancer_test.go:3842
        	Error:      	Not equal: 
        	            	expected: network.PublicIPAddress{Response:autorest.Response{Response:(*http.Response)(nil)}, Sku:(*network.PublicIPAddressSku)(nil), PublicIPAddressPropertiesFormat:(*network.PublicIPAddressPropertiesFormat)(nil), Etag:(*string)(nil), Zones:(*[]string)(nil), ID:(*string)(nil), Name:(*string)(nil), Type:(*string)(nil), Location:(*string)(nil), Tags:map[string]*string{"a":(*string)(0xc0000a0a80), "c":(*string)(0xc0000a0a90), "foo":(*string)(0xc0000a0a70), "kubernetes-cluster-name":(*string)(0xc0000a0a50), "service":(*string)(0xc0000a0a60), "y":(*string)(0xc0000a0aa0)}}
        	            	actual  : network.PublicIPAddress{Response:autorest.Response{Response:(*http.Response)(nil)}, Sku:(*network.PublicIPAddressSku)(nil), PublicIPAddressPropertiesFormat:(*network.PublicIPAddressPropertiesFormat)(nil), Etag:(*string)(nil), Zones:(*[]string)(nil), ID:(*string)(nil), Name:(*string)(nil), Type:(*string)(nil), Location:(*string)(nil), Tags:map[string]*string{"a":(*string)(0xc0000a0ab0), "c":(*string)(0xc0000a0ae0), "foo":(*string)(0xc0000a0a30), "kubernetes-cluster-name":(*string)(0xc0000a0a10), "service":(*string)(0xc0000a0a20), "y":(*string)(0xc0000a0ac0)}}
```

```
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure: TestEnsurePIPTagged expand_less | 0s
-- | --
=== RUN   TestEnsurePIPTagged --- FAIL: TestEnsurePIPTagged (0.00s)

```

#### Anything else we need to know:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

<!-- Please see the deflaking doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md) for more guidance! -->

--- a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go
@@ -135,7 +135,7 @@ func parseTags(tags string) map[string]*string {
 			klog.Warningf("parseTags: error when parsing key-value pair %s-%s, would ignore this one", k, v)
 			continue
 		}
-		formatted[k] = to.StringPtr(v)
+		formatted[strings.ToLower(k)] = to.StringPtr(v)
 	}
 	return formatted
 }

